corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js

```diff
@@ -625,24 +625,45 @@ hqDefine("cloudcare/js/form_entry/entries", function () {
         query = query.toLowerCase();
         var haystack = option.text.toLowerCase();
 
-        var match;
+        var match,
+            wordsInQuery = query.split(/\s+/),
+            wordsInChoice = haystack.split(/\s+/);
         if (matchType === Const.COMBOBOX_MULTIWORD) {
             // Multiword filter, matches any choice that contains all of the words in the query
             //
             // Assumption is both query and choice will not be very long. Runtime is O(nm)
             // where n is number of words in the query, and m is number of words in the choice
-            var wordsInQuery = query.split(' ');
-            var wordsInChoice = haystack.split(' ');
 
             match = _.all(wordsInQuery, function (word) {
                 return _.include(wordsInChoice, word);
             });
         } else if (matchType === Const.COMBOBOX_FUZZY) {
+            var isFuzzyMatch = function (haystack, query, distanceThreshold) {
+                return (
+                    (window.Levenshtein.get(haystack, query) <= distanceThreshold && query.length > 3) ||
+                    haystack === query
```

Comment:
can we change the order here to take advantage of short circuiting and avoid having to compute the levenshtein distance in some cases:
```suggestion
                    haystack === query ||
                    (query.length > 3 && window.Levenshtein.get(haystack, query) <= distanceThreshold)
```

Reply:
yep, good point